dl_fanfiction_net.py

```diff
@@ -40,12 +41,25 @@ def print_soup(soup):
     for paragraph in paragraphs:
         print(paragraph.getText())
 
+def next_chapter_exists(soup):
+    """check if page has a next chapter button"""
+    buttons = soup.find_all('button')
+    for button in buttons:
+        if button.getText() == 'Next >':
+            return True
+
 def main():
     """Prints first chapter of storyID:7241166."""
     disable_warnings()
-    url = f'{URL}s/7241166/1/Lord-Charming'
-    soup = get_beautifulsoup(url)
-    print_soup(soup)
+    chapter = 1
+    chapter_exists = True
+    while chapter_exists:
+        url = f'{URL}s/7241166/' + str(chapter) + '/Lord-Charming'
```

Comment:
```suggestion
        url = f'{URL}s/7241166/{str(chapter)}/Lord-Charming'
```
to make this less dense, I'd also suggest
```
chapter_str = str(chapter)
 url = f'{URL}s/7241166/{chapter_str}/Lord-Charming'
```
variable name is a little lackluster, so I'm good with either of the above.